app/webpacker/components/Tickets/index.jsx

```diff
@@ -93,8 +42,33 @@ function Tickets({ id }) {
     return null;
   }, [ticketDetails?.requester_stakeholders, userSelectedStakeholder]);
 
-  if (loading) return <Loading />;
-  if (error) return <Errored />;
+  const {
+    mutate: updateStatusMutate,
+    isPending: isPendingUpdateStatus,
+    isError: isErrorUpdateStatus,
+    error: errorUpdateStatus,
+  } = useMutation({
+    mutationFn: (status) => updateStatus({ ticketId: id, status, currentStakeholder }),
+    onSuccess: (status) => {
+      queryClient.setQueryData(
+        ['ticket-details', id],
+        () => ({
```

Comment:
Your solution is _almost_ perfect 😉 Instead of "grabbing" the `ticketDetails` variable from the outer scope (the return value of the `useQuery` above), the callback actually gets passed the current state of the query client as per the query key you're specifying.

So instead of...
```js
const ticketDetails = useQuery(someCoolQueryKey, queryFn);

queryClient.setQueryData(someCoolQueryKey, () => ({ ...ticketDetails, status }))
```

You should do something along the lines of...
```js
// Ignore that this variable exists! You are free to use it in your component, but not directly within the mutation success hook
const ticketDetails = { ... };

queryClient.setQueryData(someCoolQueryKey, (oldTicketDetails) => ({ ...oldTicketDetails, status }))
```

Reply:
Further reading: https://tanstack.com/query/latest/docs/reference/QueryClient#queryclientsetquerydata

Reply:
(I know, this is a very subtle change, and in your current use case it probably doesn't even make any difference. But there is a 1% chance that your current way of doing things can lead to subtle bugs when the original query result is refetched at the exact moment you're writing an update, and those bugs can be extreeeemely hard to track down and debug.)

Reply:
As a sidenote, this is considered good practice in general when using `setQueryData`. Either pass in a constant(!!) value (for example, the return value passed to the `onSuccess` hook itself can be considered constant), or pass in a function that receives the old value and recomputes it into the new value.
Never compute something based on React state (anything that comes from any function called `use...`) and pretend like it's a constant value.